skpro/tests/test_polars.py

```diff
@@ -0,0 +1,135 @@
+    reason="skip test if polars/pyarrow is not installed in environment",
+)
+def test_polars_eager_conversion_methods(
+    polars_load_diabetes_pandas, polars_load_diabetes_polars
+):
+    """
+    Tests to ensure that given a pandas dataframe, the conversion methods
+    convert properly to polars dataframe
+    """
+    X_train, X_test, y_train = polars_load_diabetes_pandas
+    X_train_pl, X_test_pl, y_train_pl = polars_load_diabetes_polars
+
+    assert check_polars_table(X_train_pl)
+    assert check_polars_table(X_test_pl)
+    assert check_polars_table(y_train_pl)
+    assert (X_train.values == X_train_pl.to_numpy()).all()
+    assert (X_test.values == X_test_pl.to_numpy()).all()
+    assert (y_train.values == y_train_pl.to_numpy()).all()
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies(["polars", "pyarrow"], severity="none"),
+    reason="skip test if polars/pyarrow is not installed in environment",
+)
+def test_polars_eager_regressor_in_fit_predict(
+    estimator, polars_load_diabetes_pandas, polars_load_diabetes_polars
+):
+    """
+    Tests to ensure that given a polars dataframe, the regression estimator
+    can fit and predict and return the correct set of outputs
+
+    Parameters
+    ----------
+
+    estimator: a given regression estimator
+
+    """
+    # create a copy of estimator to run further checks
+    estimator_copy = estimator
+    X_train, X_test, y_train = polars_load_diabetes_pandas
+    X_train_pl, X_test_pl, y_train_pl = polars_load_diabetes_polars
+
+    estimator.fit(X_train_pl, y_train_pl)
+    y_pred = estimator.predict(X_test_pl)
+
+    assert isinstance(y_pred, pl.DataFrame)
+    assert y_pred.columns == y_train_pl.columns
+    assert y_pred.shape[0] == X_test_pl.shape[0]
+
+    # code to ensure prediction values match up correctly
+    estimator_copy.fit(X_train, y_train)
+    y_pred_pd = estimator_copy.predict(X_test)
+    assert (y_pred_pd.values == y_pred.to_numpy()).all()
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies(["polars", "pyarrow"], severity="none"),
+    reason="skip test if polars/pyarrow is not installed in environment",
+)
+def test_polars_eager_regressor_in_predict_interval(
+    estimator, polars_load_diabetes_polars
+):
+    X_train_pl, X_test_pl, y_train_pl = polars_load_diabetes_polars
+
+    estimator.fit(X_train_pl, y_train_pl)
+    y_pred_interval = estimator.predict_interval(X_test_pl)
+
+    assert isinstance(y_pred_interval, pd.DataFrame)
+    assert y_pred_interval.columns[0] == ("target", 0.9, "lower")
+    assert y_pred_interval.columns[1] == ("target", 0.9, "upper")
+    assert check_raise(
+        y_pred_interval, "pred_interval", "Proba", "predict_interval return"
+    )
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies(["polars", "pyarrow"], severity="none"),
+    reason="skip test if polars/pyarrow is not installed in environment",
+)
+def test_polars_eager_regressor_in_predict_quantiles(estimator):
```

Comment:
this is failing because there is no `estimator` fixture - you need to add a `mark.parametrize` or a `pytest_generate_fixtures` implementation

Reply:
Should I specify one at random? Or is there a way to potentially pass all regression models into this test - not 100% sure what is the best approach here when specify estimators

Reply:
interesting question - for the start, I would say, pick any, as none support polars internally, later we can be more systematic